Add support for streaming hash digest calculation.

NorbertSzydlik opened the pull request.

lpil commented on src/gleam/crypto.gleam:
```diff
@@ -39,6 +39,23 @@ pub type HashAlgorithm {
+/// Starts hash algorithm for streaming
+@external(erlang, "crypto", "hash_final")
+@external(javascript, "../gleam_crypto_ffi.mjs", "hashFinal")
+pub fn hash_final(state: HashState) -> BitArray
```
Not sure about these names. How about `Hasher`, `hasher`, `hash_chunk`, and `digest`?

NorbertSzydlik replied:
Done

lpil reviewed (commented):
Very nice! Could you change the existing hash function to be implemented using this new API? And update its documentation to note the incremental API too.

We'll also need some examples in the doc comments showing how to use it, as it's a more complex API. Similar to how the stdlib shows examples. Thank you.

NorbertSzydlik commented:
> Very nice! Could you change the existing hash function to be implemented using this new API? And update its documentation to note the incremental API too.
I've added comment in docs that `hash` function uses `hash_init`, `hash_update`, and `digest`

> We'll also need some examples in the doc comments showing how to use it, as it's a more complex API. Similar to how the stdlib shows examples. Thank you.
I have added one simple example, and one complex example using `actor`.

lpil reviewed (approved):
Thank you!
